### PR TITLE
Support ProcessorJob cancel

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

This PR adds a cancel method to ProcessorJob to support canceling an unbounded stream job.

## Brief change log

- Add abstract "cancel" method to `ProcessorJob`
- Implement cancel method for `LocalProcessor` and `FlinkProcessor`
- Refactor FlinkProecssorJobs to the corresponding submitters

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: yes
## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? Python docstring

### CODE REVIEW

**Comment by Sxnan** [comment]
@lindong28 Could you review this PR?

File: python/feathub/processors/processor_job.py [hunk 1]
```diff
@@ -24,6 +24,16 @@ class ProcessorJob(ABC):
     def __init__(self) -> None:
         pass
 
+    @abstractmethod
+    def cancel(self) -> None:
```
Comment (lindong28) [inline python/feathub/processors/processor_job.py:28]: Would it useful to return a `Future` similar to `pyflink.common.JobClient.cancel()`?

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR.

File: python/feathub/processors/processor_job.py (hunk 1 above)
Comment (Sxnan) [reply python/feathub/processors/processor_job.py:28]: After looking at the `pyflink.common.JobClient`, I think the Future is only used for acknowledgment. It is completed as soon as the server confirms the cancel is received but not when the job is canceled. 

Currently, the cancel method implies the same semantics as the method should return after the cancel is acknowledged. We may need to state that in the docstring explicitly. Given that the two ways have the same semantics and we are currently not using `CompletableFuture` in other places, I am leaning toward the current way. What do you think?

Comment (Sxnan) [reply python/feathub/processors/processor_job.py]: After the offline discussion, I agree it is better to return the Future object and users can decide if they want to wait for acknowledgment or not.

**Comment by Sxnan** [comment]
@lindong28 Thank you for your review! I have updated the PR. Could you please take another look?

**Comment by lindong28** [comment]
Thanks for the update. LGTM.

**Pull request merged**